### ISSUE TITLE
Enhancement Request: Uptime Module Proxy Support and Server Probe Module Grouping & Management Features

### ISSUE THREAD

**Issue by danblaze13** [msgbyte/tianji#19]
Dear Tianji Project Maintainers,

I am writing to suggest several enhancements for the Tianji project that I believe would greatly improve the functionality and user experience of the Uptime module and the Server probe module. Below are the proposed enhancements with their respective rationales.

#### Uptime Module Enhancements:
1. **Socks Proxy Support for Uptime Monitoring:**
   - *Feature:* Add the ability to perform uptime checks through a Socks proxy, similar to the functionality offered by Uptime Kuma.
   - *Rationale:* This feature would be incredibly useful for monitoring services that are situated within internal networks. By routing requests through a Socks proxy, we can ensure that uptime monitoring covers services that are not directly accessible from the public internet, thereby providing a more comprehensive overview of service availability.

#### Server Probe Module Enhancements:
1. **Server Grouping Functionality:**
   - *Feature:* Implement a grouping feature on the Servers page that allows users to categorize servers based on different projects or criteria.
   - *Rationale:* This will help in organizing and managing servers more efficiently, especially when dealing with multiple servers across various projects. It will enhance the user interface by allowing users to navigate and monitor servers in an orderly fashion.

2. **Editable Node Names and Server Information:**
   - *Feature:* Enable the capability to edit server Node Names directly from the Servers page to increase recognizability.
   - *Feature:* Support adding remarks/tags to servers to log essential information such as billing dates or specific server details.
   - *Rationale:* These features will improve server identification and provide users with the ability to keep track of important server-related information conveniently within the platform.

3. **Network Interface Monitoring Selection:**
   - *Feature:* Allow users to select which network interface(s) to monitor, with options including 'All', a specific interface, or multiple interfaces.
   - *Rationale:* This is particularly useful for servers with multiple network cards, as it would enable accurate tracking of the actual inbound and outbound traffic. Users can tailor the monitoring to their specific setup, ensuring that the traffic statistics reflect the true network usage.

I believe that these enhancements will significantly bolster the capabilities of the Tianji project, making it a more robust and user-friendly tool for server management and uptime monitoring. I hope you will consider these suggestions for future updates.

Thank you for your time and dedication to improving the Tianji project.

---

尊敬的 Tianji 项目维护者，

我写信是为了提出对 Tianji 项目的几项增强功能建议，我相信这些改进将大大提高 Uptime 模块和服务器探针模块的功能性和用户体验。以下是提出的增强功能及其各自的理由。

#### Uptime 模块增强功能：
1. **通过 Socks 代理进行 Uptime 监控的支持：**
   - *功能：* 增加通过 Socks 代理执行 Uptime 检查的能力，类似于 Uptime Kuma 提供的功能。
   - *理由：* 对于位于内部网络中的服务，此功能非常有用。通过代理请求，我们可以确保 Uptime 监控覆盖到公共互联网直接无法访问的服务，从而提供更全面的服务可用性概览。

#### 服务器探针模块增强功能：
1. **服务器分组功能：**
   - *功能：* 在服务器页面上实现分组功能，允许用户根据不同项目或标准对服务器进行分类。
   - *理由：* 这将帮助用户更有效地组织和管理服务器，特别是在处理跨多个项目的多台服务器时。它将通过允许用户以有序的方式导航和监控服务器，增强用户界面。

2. **可编辑的节点名称和服务器信息：**
   - *功能：* 允许直接在服务器页面上编辑服务器节点名称以提高识别度。
   - *功能：* 支持为服务器添加备注/标签以记录重要信息，如账单日或特定服务器详情。
   - *理由：* 这些功能将改善服务器识别，并为用户提供在平台内方便地跟踪重要服务器相关信息的能力。

3. **网络接口监控选择：**
   - *功能：* 允许用户选择要监控的网络接口，选项包括“所有”、“特定接口”或“多个接口”。
   - *理由：* 对于具有多个网络卡的服务器，这一点特别有用，因为它将使得准确追踪实际的入站和出站流量成为可能。用户可以根据自己的具体设置来定制监控，确保流量统计反映真实的网络使用情况。

我相信这些增强功能将显著加强 Tianji 项目的能力，使其成为一个更加健壮和用户友好的服务器管理和 Uptime 监控工具。我希望您能考虑这些建议以用于未来的更新。

感谢您投入时间和精力来改进 Tianji 项目。

**Comment by moonrailgun** [comment]
yeah, thanks for your message.

I read your content in detail. and maybe some feature can be resolve now.

for network card selection, you explorer report cli (for example ./tianji_reporter --help), its support vnstat, i think its can be help you.

and you can also custom node name with params.

**Comment by danblaze13** [comment]
> yeah, thanks for your message.是的，谢谢你的留言。
> 
> I read your content in detail. and maybe some feature can be resolve now.我详细阅读了你的内容。也许现在可以解决一些功能。
> 
> for network card selection, you explorer report cli (for example ./tianji_reporter --help), its support vnstat, i think its can be help you.对于网络卡选择，您可以使用探险者报告CLI（例如./tianji_reporter --help），它支持vnstat，我认为它可以帮助您。
> 
> and you can also custom node name with params.你也可以使用参数自定义节点名称。

Yes, thank you for mentioning it. I'll try this approach to modify the detection of network cards before the feature update.

That said, I think there could be a modification of a clinical nature, for example, in the backend of the site, when copying the one-click install script, a field could be added to make it easy to change the node name all at once.

This seems to alleviate some of the duplication of effort in operations when there are more nodes.

**Comment by Muska-Ami** [comment]
+1 for "Server Grouping Functionality" and "Editable Node Names and Server Information"

**Comment by moonrailgun** [comment]
yes good idea, i have this plan to add some guide for it.(in dashboard page)

just like it:

![image](https://github.com/msgbyte/tianji/assets/6964737/86cab596-ba0a-43e1-be5f-f2edbbe1dba0)

**Comment by fistwho** [comment]
+1 for Socks Proxy Support for Uptime Monitoring
